### PR TITLE
Add more thorough documentation

### PR DESCRIPTION
**Pull request by nagitsu**
Also adds readthedocs integration.

### CODE REVIEW

File: docs/usage/quickstart.rst [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    cloud       Groups of commands to train models in the...
+    dataset     Groups of commands to manage datasets
+    eval        Evaluate trained (or training) models
+    predict     Obtain a model's predictions.
+    server      Groups of commands to serve models
+    train       Train models
+
+We'll start by downloading a checkpoint. Luminoth provides already-trained
+models so you can run predictions and get reasonable results in no time (and
+eventually be able to use for fine-tuning). In order to access these
```
Comment (dekked) [inline docs/usage/quickstart.rst]: to use for fine-tuning -> to use them for fine-tuning

File: docs/usage/quickstart.rst [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+Additional commands are available for managing checkpoints, including inspection
+and modification of checkpoints (see :ref:`cli/checkpoint`).  For now, we'll
+download a checkpoint and use it::
+
+  $ lumi checkpoint download 48ed2350f5b2
+  Downloading checkpoint...  [####################################]  100%
+  Importing checkpoint... done.
+  Checkpoint imported successfully.
+
+Once the checkpoint is downloaded, it can be used for predictions. There are
```
Comment (dekked) [inline docs/usage/quickstart.rst]: Maybe mention where the thing is stored once downloaded?

File: docs/usage/quickstart.rst [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+currently two ways to do this:
+
+- Using the CLI tool and passing it either images or videos. This will output a
+  JSON with the results and optionally draw the bounding boxes of the
+  detections in the image.
+- Using the web app provided for testing purposes. This will start a web server
+  that, when connected, allows you to upload the image. Also useful to run on
+  a remote GPU (note, however, that this is **not** production-ready).
```
Comment (dekked) [inline docs/usage/quickstart.rst]: I don't understand what is not production ready.

Seems like we could be talking about remote GPU or the web interface itself.

File: docs/usage/quickstart.rst [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+In there, you'll be able to upload an image and see the results.
+
+And that's it for the basics! Next steps would be:
+
+- Prepare your own dataset to be consumed by Luminoth (see :ref:`usage/dataset`).
+- Train a custom model with your own data, either locally or in Google Cloud
+  (see :ref:`usage/training`).
+- Turn your custom model into a checkpoint for easier sharing and usage (see
+  :ref:`usage/checkpoints`).
+- Use the Python API to call Luminoth models within Python.
```
Comment (dekked) [inline docs/usage/quickstart.rst]: Make sure we don't forget this is actually important :joy:

File: docs/usage/training.rst [hunk 5]
```diff
@@ -0,0 +1,153 @@
+.. _usage/training:
+
+Training your own model
+=======================
+
+In order to train your own model, two things are required:
+
+* An dataset ready to be consumed by Luminoth (see :ref:`usage/dataset`).
```
Comment (dekked) [inline docs/usage/training.rst]: A dataset

File: docs/usage/training.rst [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+There are a great deal of configuration options, mostly related to the model
+itself. You can, for instance, see the full range of options for the Faster
+R-CNN model, along with a brief description of each, in its `base_config.yml
+<https://github.com/tryolabs/luminoth/tree/master/luminoth/models/fasterrcnn/base_config.yml>`_
+file.
+
+Training
+--------
+
+The model training itself can either be run locally (on the CPU or GPUs
```
Comment (dekked) [inline docs/usage/training.rst]: Doesn't this sound like we support training in multiple GPUs?

File: docs/usage/training.rst [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+The ``lumi train`` CLI tool provides the following options related to training.
+
+* ``--config``/``-c``: Config file to use. If the flag is repeated, all config
+  files will be merged in left-to-right order so that every file overwrites the
+  configuration of keys defined previously.
+
+* ``--override``/``-o``: Override any configuration setting using dot notation
+  (e.g.: ``-o model.rpn.proposals.nms_threshold=0.8``).
+
+If you're using a CUDA-based GPU, you can select the GPU to use by setting the
```
Comment (dekked) [inline docs/usage/training.rst]: Add link to [NVIDIA's site](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#env-vars)?

File: docs/usage/training.rst [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+``lumi cloud gc logs`` - Fetch logs for a specific job.
+
+Options:
+  - ``--job-id``
+  - ``--polling-interval``: Seconds between each log request.
+
+Results
+```````
+
+Everything related to a job is stored on its own folder on the bucket provided
```
Comment (dekked) [inline docs/usage/training.rst]: on its own folder -> in its own folder ?

File: docs/usage/quickstart.rst (hunk 2 above)
Comment (nagitsu) [reply docs/usage/quickstart.rst]: I wouldn't add it to the quickstart section, but it should be mentioned on the checkpoint one.

**Pull request merged**